src/m4st/process_demetr.py

```diff
@@ -2,149 +2,71 @@
-
-        # Set up output arrays - typically (1000, n) where n is number of metrics
-        # Two sets of results for each metric, one fluent and one disfluent
-        mt_results = np.zeros((num_samples, len(self.metrics_to_use)))
-        dis_results = np.zeros((num_samples, len(self.metrics_to_use)))
-
-        for j, metric in enumerate(self.metrics_to_use):
-            if metric == "COMET_ref":
-                mt_results[:, j] = self.comet_ref.get_scores(
-                    ref_txts, mt_txts, src_txts
-                )
-                dis_results[:, j] = self.comet_ref.get_scores(
-                    ref_txts, dfluent_txts, src_txts
-                )
-            elif metric == "COMET_qe":
-                mt_results[:, j] = self.comet_qe.get_scores(ref_txts, mt_txts, src_txts)
-                dis_results[:, j] = self.comet_qe.get_scores(
-                    ref_txts, dfluent_txts, src_txts
-                )
-            elif metric == "BLASER_ref":
-                mt_results[:, j] = self.blaser_ref.get_scores(
-                    ref_txts, mt_txts, src_txts, blaser_lang_codes
-                )
-                dis_results[:, j] = self.blaser_ref.get_scores(
-                    ref_txts, dfluent_txts, src_txts, blaser_lang_codes
-                )
-            elif metric == "BLASER_qe":
-                mt_results[:, j] = self.blaser_qe.get_scores(
-                    mt_txts, src_txts, blaser_lang_codes
-                )
-                dis_results[:, j] = self.blaser_qe.get_scores(
-                    dfluent_txts, src_txts, blaser_lang_codes
-                )
-            elif metric == "SacreBLEU":
-                mt_results[:, j] = self.sacre_bleu.get_scores(ref_txts, mt_txts)
-                dis_results[:, j] = self.sacre_bleu.get_scores(ref_txts, dfluent_txts)
-            elif metric == "ChrF":
-                mt_results[:, j] = self.chrf.get_scores(ref_txts, mt_txts)
-                dis_results[:, j] = self.chrf.get_scores(ref_txts, dfluent_txts)
-            elif metric == "ChrF2":
-                mt_results[:, j] = self.chrf2.get_scores(ref_txts, mt_txts)
-                dis_results[:, j] = self.chrf2.get_scores(ref_txts, dfluent_txts)
-            else:
-                print(f"Unknown metric {metric}")
-
-        mask = mt_results > dis_results
-        if reverse_accuracy:
-            results = np.count_nonzero(~mask, axis=0)
-        else:
-            results = np.count_nonzero(mask, axis=0)
-
-        results = results / num_samples * 100
-
-        results_str = [category, *results]
-
-        with open(self.output_path, "a") as output_file:
-            csv_writer = csv.writer(output_file)
-            csv_writer.writerow(results_str)
+        for metric in self.metrics:  # type: ignore[has-type]
+            metric.get_scores(demetr_df, self.output_dir, cat_fp)
 
     def process_demetr(
         self,
-        samples_per_cat: int = 1000,
         cats_to_process: list | None = None,
```

Comment:
Docstring for `cats_to_process` would also be helpful (either including a list of options or linking out to where to find the list of options)

Also could be more specific with the type-hint:


```suggestion
        cats_to_process: list[str] | None = None,
```

Reply:
Added docstring